### ISSUE TITLE
Add a command line switch to allow zero exit when no tests are found.

### ISSUE THREAD

**Issue by gspencergoog** [dart-lang/test#1546]
When running with `-t tag`, it would be useful to be able to say "Yeah, if you don't find any tests with that tag, it's OK, just exit with zero instead of failing".

I want to run all of our tests twice, once with `-x tag`, and once with `-t tag` (and some other different options), but if I run it on a set of test files that don't happen to contain the tag, I'm OK with that, since they all will be run in the other test. Unfortunately, as it is now, the `-t tag` version will always fail when it doesn't find any tests, and I have to parse the test output to look for "No tests ran." and not fail in only that case (which is fragile, obviously).

/cc @natebosch

**Comment by gspencergoog** [comment]
More is here: https://github.com/dart-lang/test/issues/1539#issuecomment-867086151

**Comment by jakemac53** [comment]
Would this satisfy the use case as well? https://github.com/dart-lang/test/issues/1418

tldr; provide a different exit code, so you could just check for that

**Comment by gspencergoog** [comment]
Yeah, that would probably work: anything we can use to distinguish would work, as long as it doesn't collide with other exit codes we might get.
